File: 230-二叉搜索树中第K小的元素.cpp
class Solution {
public:
    int kthSmallest(TreeNode* root, int k) {
        stack<TreeNode*> mystack;
        TreeNode* cur_node = root;
        int result;

        for(int i = 0; i < k; i++){
            while(cur_node){
                mystack.push(cur_node);
                cur_node = cur_node->left;
            }
            cur_node = mystack.top();
            mystack.pop();
            result = cur_node->val;
            cur_node = cur_node->right;
        }

        return result;
    }
};

File: 230-二叉搜索树中第K小的元素.py
class Solution:
    def kthSmallest(self, root: TreeNode, k: int) -> int:
        mystack = []
        cur_node = root
        
        for i in range(k):
            while cur_node:
                mystack.append(cur_node)
                cur_node = cur_node.left
            cur_node = mystack.pop()
            result = cur_node.val
            cur_node = cur_node.right
        
        return result
